File: ByLanguage/android/Cookbook/PassingData/app/src/main/java/com/tq/passingdata/MainActivity.java
package com.tq.passingdata;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.widget.Button;
import android.widget.TextView;
import android.content.Intent;

public class MainActivity extends AppCompatActivity {

        private TextView label;

        @Override
        protected void onCreate(Bundle savedInstanceState) {
                super.onCreate(savedInstanceState);
                setContentView(R.layout.activity_main);

                label = (TextView) findViewById(R.id.label);
                Button button = (Button) findViewById(R.id.button);
                button.setOnClickListener((view) -> {
                                Intent intent = new Intent(this, SecondActivity.class);
                                intent.putExtra(SecondActivity.FIRST_PARAM, 1);
                                intent.putExtra(SecondActivity.SECOND_PARAM, 2);
                                startActivityForResult(intent, SecondActivity.SUM);
                        });
        }

        protected void onActivityResult(int requestCode, int resultCode, Intent payload) {

                if (requestCode == SecondActivity.SUM) {
                        if (resultCode == RESULT_OK) {
                                int result = payload.getIntExtra(SecondActivity.RESULT, 0);
                                label.setText("Result: " + result);
                        }
                }
        }
}

File: ByLanguage/android/TimePickerDemo/app/src/main/java/com/tq/timepickerdemo/MainActivity.kt
package com.tq.timepickerdemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.TimePicker

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val timePicker1 = findViewById<TimePicker>(R.id.time_picker1)
    }
}
